packages/beyondinsight_password_safe/data_stream/useraudit/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,164 @@
+              }
+            }
+
+            k = regex.matcher(k).replaceAll('$1_$2').toLowerCase();
+            snakeCaseMap.put(k, v);
+          }
+          return snakeCaseMap;
+        }
+
+        if (ctx.beyondinsight_password_safe.useraudit != null) {
+          ctx.beyondinsight_password_safe.useraudit = keysToSnakeCase(ctx.beyondinsight_password_safe.useraudit);
+        }
+
+##################### ECS Fields #####################
+  - set:
+      field: ecs.version
+      value: '8.11.0'
+
+  - set:
+      field: event.kind
+      value: event
+
+  - append:
+      field: event.category
+      value: iam
+
+  - append:
+      field: event.type
+      value: info
+
+  - set:
+      field: event.dataset
+      value: beyondinsight_password_safe.useraudit
+
+  - set:
+      field: event.module
+      value: beyondinsight_password_safe
+
+  - set:
+      field: event.id
+      value: "{{{beyondinsight_password_safe.useraudit.audit_id}}}"
+      ignore_empty_value: true
+
+  - set:
+      field: user.id
+      value: "{{{beyondinsight_password_safe.useraudit.user_id}}}"
+      ignore_empty_value: true
+
+  - append:
+      field: related.user
+      value: "{{{beyondinsight_password_safe.useraudit.user_id}}}"
+      allow_duplicates: false
+
+  - set:
+      field: user.name
+      value: "{{{beyondinsight_password_safe.useraudit.user_name}}}"
+      ignore_empty_value: true
+
+  - append:
+      field: related.user
+      value: "{{{beyondinsight_password_safe.useraudit.user_name}}}"  
+      allow_duplicates: false
+
+  - append:
+      field: related.hosts
+      value: "{{{beyondinsight_password_safe.useraudit.ipaddress}}}"
+      allow_duplicates: false
+ #################### Override host.ip ######################
+  - script:
+      lang: painless
+      source: |
+         def ipstring = ctx.beyondinsight_password_safe.useraudit.ipaddress.toString().splitOnToken(".");
+         def ip = String.valueOf(Integer.parseInt(ipstring[0]))+"."+String.valueOf(Integer.parseInt(ipstring[1]))+"."+String.valueOf(Integer.parseInt(ipstring[2]))+"."+String.valueOf(Integer.parseInt(ipstring[3]));
+         ctx.beyondinsight_password_safe.useraudit.ipaddress = ip;
+
+  - convert:
+      field: beyondinsight_password_safe.useraudit.ipaddress
+      target_field: beyondinsight_password_safe.useraudit.ipaddress
+      type: ip
+      if: ctx.beyondinsight_password_safe.useraudit?.ipaddress != null && ctx.beyondinsight_password_safe.useraudit.ipaddress != ''
+      ignore_failure: true 
+      ignore_missing: true
+
+  - append:
+      field: host.ip
+      value: "{{{beyondinsight_password_safe.useraudit.ipaddress}}}"
+
+  - geoip:
+      field: host.ip
+      target_field: host.geo
+      ignore_missing: true
+      ignore_failure: true
+################### Set @timestamp ##########################
+  - date:
+     field: "beyondinsight_password_safe.useraudit.create_date"
+     target_field: "@timestamp"
+     formats: ["ISO8601"]
```

You can give an ISO-formatted timestamp to Elasticsearch for a field of `date` type. The create date is already in ISO 8601 format, so you can just copy the value.
```suggestion
  - set:
     field: "@timestamp"
     copy_from: beyondinsight_password_safe.useraudit.create_date
     ignore_empty_value: true
```